DBSCE/Overview.md

```diff
@@ -0,0 +1,275 @@
+- The tokens can be delivered to the RP directly through an API instead of a header based response.
+- The tokens also contain the thumbbrint of the public key, and the RP must validate the thumbprint from the token against the thumbprint from the JWT. The RP will also embed the thumbprint in the cookie for subsequent session validation.
+
+#### IDP Calls Public Local Key Helper
+
+In many use cases, it is also a valid to have separate servers as [RP](#relying-party-rp) and [IdP](#identity-provider-idp). We address the use case where the IdP calls a _public Local Key Helper_ and is a separate service from the RP below.
+
+For easy mapping with the existing DBSC proposal, please note:
+
+- Steps 1-16 specify the key generation process for a public local key helper.
+- Steps 17-29 are [DBSC](https://github.com/wicg/dbsc), added for completeness.
+
+
+![IDPCallsPublicLocalKeyHelper](./IDPCallsPublicLocalKeyHelper.svg)
+
+#### IDP Calls Private Local Key Helper
+
+As referred in [Local Key Helper](#local-key-helper) section, it is possible for an IdP to implement a private local key helper and establish a private protocol to communicate. Any such private local key helper can only be used by the specific IDP that owns its implementation.
+
+In DBSC(E), the use of private local key helper for specific IDPs enables the below optimizations:
+
+- Since the IDP can trust the private local key helper, the `HelperId` discovery can be optimized by caching. An IDP can specify its preferred ordered list of `HelperId`s in any response, and can request the browser to cache its preference for a specific duration(`HelperCacheTime`). Also, such requests need not be initiated by the RP that implements DBSC(E), the IDP can specify its `HelperIdList` in any communication(request/response) within the browser context.
+- The IDP and the private local key helper can also leverage `long lived` binding statements. If the IdP can establish the proof of presence of the device by its own proprietery mechanism (usually authentication) for a given request, then it can accept binding statements without requiring fresh `nonce` challenge for those. Once the device authentication is complete, the IDP can use the long lived binding statements for binding proof. Browser can cache those binding statements for further performance optimization.
+- RP can combine the request for a DBSC(E) session as a part of IDP redirection during sign-in. If the key generation and sign in are successful, the browser can intercept the sign-in flow to generate (with local key helper, which is private to IdP) and append the new JWT proof to the response, before navigating to RP. This allows the RP to avoid an additional round trip to initiate the DBSC(E) session.
+
+
+![IDPCallsPrivateLocalKeyHelper](./IDPCallsPrivateLocalKeyHelper.svg)
```

Comment:
I think at some point we also discussed refresh optimization, but I think we didn't finalize it. Maybe we can update it as a follow up to the first version of the explainer. 

From notes:

>Refresh Session Optimization: During the meeting, the team agreed to implement refresh session optimization for inactive sessions. This involves adding an optional signal to indicate that workload requests will include additional headers with JWTs to refresh the authentication cookie. Kristian is responsible for updating the DBSC with this.

Reply:
We have clarified this as indicated by defer_requests=true in the original DBSC.